supriya/tools/ugentools/Duty.py

```diff
@@ -0,0 +1,265 @@
+# -*- encoding: utf-8 -*-
+from supriya.tools.ugentools.UGen import UGen
+
+
+class Duty(UGen):
+    r"""
+    A value is demanded of each UGen in the list and output according to a stream of duration values.
+
+    ::
+        >>> duty = ugentools.Duty.kr(
+        ...     done_action=0,
+        ...     duration=ugentools.Drand(
+        ...         sequence=[0.01, 0.2, 0.4],
+        ...         repeats=2,
+        ...     ),
+        ...     reset=0,
+        ...     level=ugentools.Dseq(
+        ...         sequence=[204, 400, 201, 502, 300, 200],
+        ...         repeats=2,
+        ...         ),
+        ...     )
+        >>> duty
+        Duty.kr()
+
+    """
+
+    # ### CLASS VARIABLES ###
+
+    __documentation_section__ = None
+
+    __slots__ = ()
+
+    _ordered_input_names = (
+        'duration',
+        'reset',
+        'level',
+        'done_action',
+    )
+
+    _valid_calculation_rates = None
+
+    # ### INITIALIZER ###
+
+    def __init__(
+            self,
```

Comment:
I let PyCharm format the code amd reverted some changes by hand. So I guess the answer is it is too late and  I should sleep :)

I'll look into yapf in the morning. I would really like to have  a tool autoformat my code and have a consistent style without worrying about it too much.